gpflow/conditionals/mo_conditionals.py

```diff
@@ -50,8 +50,7 @@ def shared_independent_conditional(
     """
     Kmm = covariances.Kuu(inducing_variable, kernel, jitter=default_jitter())  # [M, M]
     Kmn = covariances.Kuf(inducing_variable, kernel, Xnew)  # [M, N]
-    Knn = kernel(Xnew, full=full_cov, full_output_cov=False)
-    Knn = Knn[0, ...] if full_cov else Knn[..., 0]  # [N, N] or [N]
+    Knn = kernel.kernel(Xnew, full=full_cov)
```

Comment:
There was an inefficiency in the code in which the SharedIndependent kernel used tile() to construct a [P, N, N] matrix, of which this conditional implementation then selected the first slice. Instead, we can simply call the base kernel and only construct the [N, N] matrix.

Reply:
is this related to the PR title?

Reply:
not really, but it's a good catch.